src/Components/Blazor/Build/src/targets/Blazor.MonoRuntime.targets

```diff
@@ -129,25 +109,33 @@
       Condition="'$(BlazorLinkOnBuild)' != 'true' AND '$(BlazorLinkOnBuild)' != 'false'" />
 
     <ItemGroup>
-      <!--
-        ReferenceCopyLocalPaths includes all files that are part of the build out with CopyLocalLockFileAssemblies on.
-        Remove assemblies that are inputs to calculating the assembly closure. Instead use the resolved outputs, since it is the minimal set.
-       -->
-      <_BlazorCopyLocalPaths Include="@(ReferenceCopyLocalPaths)" />
-      <_BlazorCopyLocalPaths Remove="@(_BlazorManagedRuntimeAssemby)" />
-
-      <BlazorOutputWithTargetPath Include="@(_BlazorCopyLocalPaths)">
-        <BlazorRuntimeFile>true</BlazorRuntimeFile>
-        <TargetOutputPath>$(BlazorRuntimeBinOutputPath)%(_BlazorCopyLocalPaths.DestinationSubDirectory)%(FileName)%(Extension)</TargetOutputPath>
-        <RelativeOutputPath>%(_BlazorCopyLocalPaths.DestinationSubDirectory)%(FileName)%(Extension)</RelativeOutputPath>
-      </BlazorOutputWithTargetPath>
```

Comment:
@pranavkm I deleted this whole block because it seems like it's no longer required now we resolve things through the linker. Additionally, it was causing problems:

 1. It was causing double-writes for the pdb files
 2. It was causing runtime errors in the hash checking, because one of the copies of the PDBs it writes is the "pre-linked" version so if you had linking enabled, the hashes didn't match up with the post-linked copies.

Do you agree this logic was redundant and could be removed?

Reply:
> it seems like it's no longer required now we resolve things through the linker

Not when when `BlazorLinkOnBuild=false`. It's still using our magical task.

Reply:
It does still seem to work fine like this even with BlazorLinkOnBuild=false. Do you have any idea how I might be able to repro a problem caused by removing this block?

Reply:
OK, I now understand what problem is caused by removing this (it breaks the previous handling of satellite assemblies).

However I am not at all sure that the previous handling of satellite assemblies is what we currently want. Rather than tie these two concerns together, I'm trying to retain the existing satellite assembly handling and keep its tests passing, and have filed https://github.com/dotnet/aspnetcore/issues/18951 to follow up on that feature separately.

As such, this block is now partially reintroduced.